validphys2/src/validphys/process_options.py

```diff
@@ -32,29 +32,34 @@ class _Vars:
     m_ttBar = "m_ttBar"
 
 
-def _map_to_metadata(kin_df, metadata):
+class _KinematicsInformation:
     """Read the 3 columns dataframe corresponding to the values set in the
     ``kinematic_coverage`` field into a dictionary defining the name of the variables.
     Adds the special "sqrts" key unless it is already part of the kinematic coverage.
+
+    Provides a ``.get_one_of`` method that accept any number of variables
     """
-    kin_cov = metadata.kinematic_coverage
-    kins = {}
-    for kin_lab, kin_values in zip(kin_cov, kin_df.values.T):
-        kins[kin_lab] = kin_values
 
-    if "sqrts" not in kin_cov:
-        kins[_Vars.sqrts] = metadata.cm_energy
+    def __init__(self, kin_df, metadata):
+        kin_cov = metadata.kinematic_coverage
+        kins = {}
+        for kin_lab, kin_values in zip(kin_cov, kin_df.values.T):
+            kins[kin_lab] = kin_values
+
+        if "sqrts" not in kin_cov:
+            kins[_Vars.sqrts] = metadata.cm_energy
 
-    return kins
+        self._kins = kins
+        self._variables = list(kins.keys())
 
+    def get_one_of(self, *variables):
```

Comment:
```suggestion
    def get_one_of(self, *variables):
```
Please add a docstring explaining the purpose of this function is to be able to generalize the xQ2 mappings for various datasets.